From 0d2f6b39b1909e7af4144516d426e2d740e665bc Mon Sep 17 00:00:00 2001
From: Zhong Hongbo <hongbo.zhong@windriver.com>
Date: Fri, 10 Oct 2014 17:09:58 +0800
Subject: [PATCH 1817/1825] mv_hal: Fix the compile warning

commit 1e6a57597687 [crypto: mvebu_cesa: Move features options] from
https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
introduces the following compile warning:

1. Fix the warning: "CONFIG_OF" is not defined
2. Fix the warning: ISO C90 forbids mixed declarations and code
3. Fix the warning: initialization from incompatible pointer type

Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 .../plat-armada/mv_drivers_lsp/mv_cesa/cesa_dev.c  |    4 ++++
 arch/arm/plat-armada/mv_hal/cesa/mvCesa.c          |    4 ++--
 arch/arm/plat-armada/mv_hal/cntmr/mvCntmr.c        |    3 ++-
 3 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/arch/arm/plat-armada/mv_drivers_lsp/mv_cesa/cesa_dev.c b/arch/arm/plat-armada/mv_drivers_lsp/mv_cesa/cesa_dev.c
index d0e1df8..f40a0dd 100644
--- a/arch/arm/plat-armada/mv_drivers_lsp/mv_cesa/cesa_dev.c
+++ b/arch/arm/plat-armada/mv_drivers_lsp/mv_cesa/cesa_dev.c
@@ -156,7 +156,11 @@ int run_cesa_debug(CESA_DEBUG *cesa_debug)
 }
 
 
+#ifdef HAVE_UNLOCKED_IOCTL
+static long
+#else
 static int
+#endif
 cesadev_ioctl(
 	struct file *filp,
 	unsigned int cmd,
diff --git a/arch/arm/plat-armada/mv_hal/cesa/mvCesa.c b/arch/arm/plat-armada/mv_hal/cesa/mvCesa.c
index b39a8c5..4d3f097 100644
--- a/arch/arm/plat-armada/mv_hal/cesa/mvCesa.c
+++ b/arch/arm/plat-armada/mv_hal/cesa/mvCesa.c
@@ -186,9 +186,9 @@ static INLINE void mvCesaReqProcessStart(MV_U8 chan, MV_CESA_REQ *pReq)
 {
 	MV_32 frag;
 
-#ifdef MV_CESA_CHAIN_MODE
+#if defined(MV_CESA_CHAIN_MODE)
 	pReq->state = MV_CESA_CHAIN;
-#elif CONFIG_OF
+#elif defined(CONFIG_OF)
 	if (mv_cesa_feature == CHAIN)
 		pReq->state = MV_CESA_CHAIN;
 	else
diff --git a/arch/arm/plat-armada/mv_hal/cntmr/mvCntmr.c b/arch/arm/plat-armada/mv_hal/cntmr/mvCntmr.c
index 956beb8..98b08f2 100755
--- a/arch/arm/plat-armada/mv_hal/cntmr/mvCntmr.c
+++ b/arch/arm/plat-armada/mv_hal/cntmr/mvCntmr.c
@@ -446,12 +446,13 @@ MV_STATUS mvCntmrIntClear(MV_U32 cntmrNum)
 *******************************************************************************/
 MV_U32 mvCntmrFrqGet(MV_U32 cntmrNum)
 {
+	MV_U32 cntmrCtrl;
+
 	if (cntmrHalData.ctrlFamily==MV_78XX0) 
 	{
 		if (INVALID_CNTMR(cntmrNum))
 			return MV_ERROR;
 
-		MV_U32 cntmrCtrl;
 		cntmrCtrl = MV_REG_READ(CNTMR_CTRL_REG(cntmrNum));
 		if (cntmrCtrl & CTCR_ARM_TIMER_25MhzFRQ_MASK(cntmrNum))
 			return MV_BOARD_REFCLK_25MHZ;                
-- 
1.7.5.4

